src/passes/OptimizeInstructions.cpp

```diff
@@ -1985,6 +1985,71 @@ struct OptimizeInstructions
         c->value = Literal::makeZero(c->type);
         return;
       }
+      // Prefer compare to signed min (s_min) instead of s_min + 1.
+      // (signed)x < s_min + 1   ==>   x == s_min
+      if (binary->op == LtSInt32 && c->value.geti32() == -2147483647L) {
```

Comment:
How about using `INT32_MIN + 1` here and similar constants instead of hardcoded numbers elsewhere?

Reply:
Good idea!